extensions/panache/mongodb-panache-common/deployment/src/test/java/io/quarkus/mongodb/panache/common/MongoDatabaseResolverTest.java

```diff
@@ -130,12 +131,12 @@ private void assertPerson(final Person expected, final Person value) {
     private static class CustomMongoOperations extends MongoOperations<Object, PanacheUpdate> {
 
         @Override
-        protected Object createQuery(MongoCollection<?> collection, ClientSession session, Document query, Document sortDoc) {
+        protected Object createQuery(MongoCollection<?> collection, ClientSession session, Bson query, Bson sortDoc) {
             return null;
         }
 
         @Override
-        protected PanacheUpdate createUpdate(MongoCollection collection, Class<?> entityClass, Document docUpdate) {
+        protected PanacheUpdate createUpdate(MongoCollection collection, Class<?> entityClass, Bson docUpdate) {
```

Comment:
Yes, under the cover it uses PanacheUpdateImpl that was already using `Bson` in its signature. Always better to use interfaces instead of implementation type.

Reply:
Well, in this case, the interface is low level and the `Document` has a real semantic (it's a Document, not some random Bson object).

Anyway, I won't oppose to it, I was just asking.